onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -713,12 +726,12 @@ def _infer_Slice(self, node):
                     new_shape[i] = self._new_symbolic_dim_from_output(node,0,i)
         else:
             for i,s,e,t in zip(axes, starts, ends, steps):
-                # TODO: handle step
-                assert t == 1
                 idx = handle_negative_axis(i, len(new_shape))
                 if is_literal(e):
                     if e >= int(2 ** 31 - 1): # max value of int32
                         e = new_shape[i]
+                    elif e <= -int(2 ** 31 - 1):
```

Comment:
Are we checking ```e``` against ```MIN_INT32```? If so, probably it should be ```int(-(2 ** 31))```?

Reply:
Sure